### ISSUE TITLE
Loss function

### ISSUE THREAD

**Issue by PigUnderRoof** [yuyangw/MolCLR#4]
https://github.com/yuyangw/MolCLR/blob/d1ffa650733267f06637e859b10cd09de02e2736/utils/nt_xent.py#L62

Hi @yuyangw, cool pre-training task design, nice work!

Please help me understand this:

Labels for pos and neg samples are both 0 in loss function module, and I failed to find sign flipping for pos prediction. Why does this make sense, or am I missing something here?

**Comment by yuyangw** [comment]
Hi, thank you for the kind words!

You can think of the contrastive loss as a 2N-1 way classification. Given a latent vector z, you have another latent vector from a positive instance and 2N-2 other latent vectors from negative instances. You need to correctly find the label of z, which is the positive instance, from the total 2N-1 instances. Thus, the target label of z is always zero, which is exactly the index of the positive instance. 

Hope this helps!

Best
